System/Console/ANSI/Color.hs

```diff
@@ -0,0 +1,102 @@
+-- | Conversions between color representations
+module System.Console.ANSI.Color
+  (
+  -- | 8-bit to 24-bit color conversion
+    color8To24
+  -- | Utility functions
+  , clamp
+  , color8CodeToEither
+  , closest4bitsANSIColor
+    ) where
+
+
+import qualified Data.Colour.SRGB.Linear as SRGBLin(rgb, Colour)
+import qualified Data.Colour.SRGB as SRGB (RGB (..), toSRGB, Colour)
+
+import Data.Colour.Names (black, blue, cyan, green, grey, lime, magenta, maroon,
+                          navy, olive, purple, red, silver, teal, white, yellow)
+
+import Data.List (minimumBy, (!!))
+
+import System.Console.ANSI.Types
+
+
+color8To24 :: Color8 -> SRGBLin.Colour Float
+color8To24 (RGB8Color r' g' b') =
+  let mapping = (\v -> v/255.0) . fromIntegral . xtermMapRGB8bitComponent . (\v -> clamp v 0 5) :: Int -> Float
+      r = mapping r'
+      g = mapping g'
+      b = mapping b'
+  in  SRGBLin.rgb r g b
+color8To24 (Gray8Color g') =
+  let mapping = (\v -> v/255.0) . fromIntegral . xtermMapGray8bitComponent . (\v -> clamp v 0 23) :: Int -> Float
+      g = mapping g'
+  in  SRGBLin.rgb g g g
+
+-- | how xterm interprets 8bit rgb colors (deduced from https://jonasjacek.github.io/colors/)
+{-# INLINE xtermMapRGB8bitComponent #-}
+xtermMapRGB8bitComponent :: Int -> Int
+xtermMapRGB8bitComponent 0 = 0
+xtermMapRGB8bitComponent n = 95 + (pred n) * 40
```

Comment:
Perhaps expressed more simply as the equivalent:
`xtermMapRGB8bitComponent n = 55 + n * 40`

Reply:
good catch!